homeassistant/components/media_player/apple_tv.py

```diff
@@ -151,6 +159,22 @@ def _has_playing_media_changed(self, new_playing):
         return new_playing.media_type != old_playing.media_type or \
             new_playing.title != old_playing.title
 
+    def playstatus_error(self, updater, exception):
+        """Inform about an error and restart push updates."""
+        if isinstance(exception, _base.TimeoutError):
+            _LOGGER.warning(
+                'Timed out while connecting to device, trying again')
+        else:
+            _LOGGER.warning('A %s error occurred: %s',
+                            exception.__class__, exception)
+
+        # This will wait 10 seconds before restarting push updates. If the
+        # connection continues to fail, it will flood the log (every 10
+        # seconds) until it succeeds. A better approach should probably be
+        # implemented here later.
+        updater.start(initial_delay=10)
+        self._hass.async_add_job(self.async_update_ha_state())
```

Comment:
What, I must have missed something here. It's supposed to reset ``self._playing``. I'll add that, good spotting!